org.freedesktop.Sdk.Extension.haskell.json

```diff
@@ -0,0 +1,155 @@
+                {
+                    "type": "archive",
+                    "only-arches": [
+                        "x86_64"
+                    ],
+                    "url": "https://downloads.haskell.org/~ghc/8.10.2/ghc-8.10.2-x86_64-deb10-linux.tar.xz",
+                    "x-checker-data": {
+                        "type": "html",
+                        "url": "https://www.haskell.org/ghc/download.html",
+                        "version-pattern": "((?:\\d+\\.)(?:\\d+\\.)(?:\\d+))",
+                        "url-template": "https://downloads.haskell.org/~ghc/$version/ghc-$version-x86_64-deb10-linux.tar.xz"
+                    },
+                    "sha256": "94513d82c38c848f489113a75fa5ef4e5a8e3ecfaa74ca90e2620d2193ff1632",
+                    "size": 213280584
+                }
+            ]
+        },
+        {
+            "name": "cabal",
+            "buildsystem": "simple",
+            "only-arches": [
+                "x86_64"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "only-arches": [
+                        "x86_64"
+                    ],
+                    "url": "https://downloads.haskell.org/~cabal/cabal-install-3.2.0.0/cabal-install-3.2.0.0-x86_64-unknown-linux.tar.xz",
+                    "strip-components": 0,
+                    "x-checker-data": {
+                        "type": "html",
+                        "url": "https://www.haskell.org/cabal/download.html",
+                        "version-pattern": "version ((?:\\d+\\.)?(?:\\d+\\.)?(?:\\d+\\.)?(?:\\d+))",
+                        "url-template": "https://downloads.haskell.org/~cabal/cabal-install-$version/cabal-install-$version-x86_64-unknown-linux.tar.xz"
+                    },
+                    "sha256": "32d1f7cf1065c37cb0ef99a66adb405f409b9763f14c0926f5424ae408c738ac",
+                    "size": 4363200
+                }
+            ],
+            "build-commands": [
+                "install -C ./cabal ${FLATPAK_DEST}/bin/"
+            ]
+        },
+        {
+            "name": "stack",
+            "buildsystem": "simple",
+            "only-arches": [
+                "x86_64"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "only-arches": [
+                        "x86_64"
+                    ],
+                    "url": "https://github.com/commercialhaskell/stack/releases/download/v2.3.3/stack-2.3.3-linux-x86_64.tar.gz",
+                    "x-checker-data": {
+                        "type": "html",
+                        "url": "https://github.com/commercialhaskell/stack/releases/latest",
+                        "version-pattern": "/commercialhaskell/stack/releases/tag/v((?:\\d+\\.)?(?:\\d+\\.)?(?:\\d+))",
+                        "url-template": "https://github.com/commercialhaskell/stack/releases/download/v$version/stack-$version-linux-x86_64.tar.gz"
```

Comment:
Here's the offending line:

https://github.com/flathub/flatpak-external-data-checker/blob/05a99d4cb6aaa924b7b03f31389e6eebde19af77/src/checkers/htmlchecker.py#L102

Some of the other checkers (always or optionally, I can't remember) replace the URL with the original one, you could make this a parameter of the html checker, or just change its default behaviour, though I bet there are now some projects which rely on the current behaviour. It's all rather ad-hoc :(